app/templates/views/broadcast/partials/area-map-postcodes.html

```diff
@@ -1,66 +1,142 @@
 <script src="{{ asset_url('javascripts/leaflet.js') }}"></script>
+<script src="{{ asset_url('javascripts/proj4.js') }}"></script>
+
 <script>
-  let polygons = []
-
-  {% if broadcast_message.areas %}
-    {% for polygon in broadcast_message.simple_polygons_with_bleed.as_coordinate_pairs_lat_long %}
-      polygons.push(
-        L.polygon({{polygon}}, {
-          opacity: 1,
-          color: '#1d70b8',
-          fillColor: '#5694ca',
-          fillOpacity: 0.15,
-          weight: 2,
-          dashArray: [6, 7],
-          lineCap: 'butt'
-        })
-      );
-    {% endfor %}
-
-    {% for polygon in broadcast_message.polygons.as_coordinate_pairs_lat_long %}
-      polygons.push(
-        L.polygon({{polygon}}, {
-          color: '#0b0b0c',
-          fillColor: '#5694ca',
-          fillOpacity: 0.3,
-          weight: 2
-        })
-      );
-      let bounds = L.polygon({{polygon}}).getBounds();
-      let center = bounds.getCenter();
-
-    {% endfor %}
-  {% endif %}
 
+  let pinpoint
+  let circle
+  const markerIcon = L.icon({
+    iconUrl:"{{ asset_url('images/marker-icon.png') }}",
+    iconSize: [30, 30],
+  });
 
   let mapElement = document.getElementById('area-list-map');
-  mapElement.style.height = Math.max(320, window.innerHeight - mapElement.offsetTop - 100) + 'px';
-  const mymap = L.map(
+  mapElement.style.height = Math.max(400, window.innerHeight - mapElement.offsetTop - 100) + 'px';
+  let radius = document.getElementById('radius').value;
+  let area_list = document.querySelector('.area-list');
+
+  function getCoordinateType() {
+    const pageUrl = window.location.href;
+    if (pageUrl.includes('cartesian')) {
+      return 'cartesian';
+    } else if (pageUrl.includes('decimal')) {
+      return 'decimal';
+    } else if (pageUrl.includes('postcodes')) {
+      return 'postcode';
+    }
+  }
```

Comment:
This feels like a fairly fragile way of determining what type of page we're on.  The partial already has dynamic content so why not let the server decide.  If you still needed it and didn't want to create dynamic javascript you could always create a small hidden field with an id which you could then fetch the value of (which you're going to need when you submit the form anyway).

Reply:
I've removed this and instead use the coordinate_type variable passed in for coordinates and for postcodes it checks for a variable only set when map is rendered on postcode page.